internal/cmd/skupper/site/kube/site_delete.go

```diff
@@ -0,0 +1,113 @@
+package kube
+
+import (
+	"context"
+	"fmt"
+	"github.com/skupperproject/skupper/client"
+	"github.com/skupperproject/skupper/internal/cmd/skupper/utils"
+	skupperv1alpha1 "github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/typed/skupper/v1alpha1"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	siteDeleteLong = `Delete a site by name`
+)
+
+type CmdSiteDelete struct {
+	Client    skupperv1alpha1.SkupperV1alpha1Interface
+	CobraCmd  cobra.Command
+	Namespace string
+	siteName  string
+}
+
+func NewCmdSiteDelete() *CmdSiteDelete {
+
+	skupperCmd := CmdSiteDelete{}
+
+	cmd := cobra.Command{
+		Use:     "delete",
+		Short:   "Delete a site",
+		Long:    siteDeleteLong,
+		Example: "skupper site delete my-site",
+		PreRun:  skupperCmd.NewClient,
+		Run: func(cmd *cobra.Command, args []string) {
+			utils.HandleErrorList(skupperCmd.ValidateInput(args))
+			utils.HandleError(skupperCmd.Run())
+		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			return skupperCmd.WaitUntilReady()
+		},
+	}
+
+	skupperCmd.CobraCmd = cmd
+
+	return &skupperCmd
+}
+
+func (cmd *CmdSiteDelete) NewClient(cobraCommand *cobra.Command, args []string) {
+	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), "")
+	utils.HandleError(err)
+
+	cmd.Client = cli.GetSkupperClient().SkupperV1alpha1()
+	cmd.Namespace = cli.Namespace
+}
+
+func (cmd *CmdSiteDelete) AddFlags() {}
+
+func (cmd *CmdSiteDelete) ValidateInput(args []string) []error {
+	var validationErrors []error
+
+	//Validate if there is already a site defined in the namespace
+	siteList, err := cmd.Client.Sites(cmd.Namespace).List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	} else if siteList == nil || (siteList != nil && len(siteList.Items) == 0) {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no existing Skupper site resource to update"))
+	} else if len(siteList.Items) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("there are several sites in this namespace and and it should be only one"))
```

Comment:
If no site name is provided, then this beahivor makes sense, but if user enters the site name to
delete, then I understand it should delete the specified one

Reply:
done! bd679654bdad99d6f2e64d2b0a4158fcdea3e02e